File: AspNetMvcUnleashed/0672329980_source/Chapter06/VB/MvcApplication1/MvcApplication1/Controllers/CustomerController.vb

Public Class CustomerController
    Inherits System.Web.Mvc.Controller

    Private _entities As New CustomersDBEntities()

    Function Index() As ActionResult
        ViewData("CustomerId") = New SelectList(_entities.CustomerSet.ToList(), "Id", "LastName")
        Return View()
    End Function

    Function Register() As ActionResult
        Return View()
    End Function
    

    Function Create() As ActionResult
        Return View()
    End Function
  
    Function Delete() As ActionResult
        Return View()
    End Function

    Function List() As ActionResult
        ViewData("Customers") = From c In _entities.CustomerSet _
                                Select c.LastName
        Return View()
    End Function

End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter06/CS/MvcApplication1/MvcApplication1/Controllers/CustomerController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using System.Web.Mvc.Ajax;
using MvcApplication1.Models;

namespace MvcApplication1.Controllers
{
    public class CustomerController : Controller
    {
        private CustomersDBEntities _entities = new CustomersDBEntities();

        public ActionResult Index()
        {
            ViewData["CustomerId"] = new SelectList(_entities.CustomerSet.ToList(), "Id", "LastName");
            return View();
        }


        public ActionResult Create()
        {
            return View();
        }


        public ActionResult Register()
        {
            return View();
        }


        public ActionResult Details()
        {
            return View();
        }


        public ActionResult Delete()
        {
            return View();
        }


        public ActionResult List()
        {
            ViewData["Customers"] = from c in _entities.CustomerSet
                                    select c.LastName; 
            return View();
        }


    }
}
